packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol

```diff
@@ -23,105 +22,222 @@ import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+
+    uint256 internal _anchorGameCandidateIndex;
+
+    IFaultDisputeGame internal _anchorGame;
+
+    uint256 public tryUpdateAnchorStateGas;
+
+    constructor() {
         _disableInitializers();
     }
 
     /// @notice Initializes the contract.
-    /// @param _startingAnchorRoots An array of starting anchor roots.
+    /// @param _disputeGameFactory DisputeGameFactory address.
+    /// @param _disputeGameFinalityDelaySeconds The delay between when a dispute game is resolved and when it can be
+    /// considered finalized.
+    /// @param _authorizedGame An authorized dispute game.
     /// @param _superchainConfig The address of the SuperchainConfig contract.
+    /// @param _tryUpdateAnchorStateGas The approximate gas limit for the tryUpdateAnchorState loop.
     function initialize(
-        StartingAnchorRoot[] memory _startingAnchorRoots,
-        ISuperchainConfig _superchainConfig
+        IDisputeGameFactory _disputeGameFactory,
+        uint256 _disputeGameFinalityDelaySeconds,
+        IFaultDisputeGame _authorizedGame,
+        ISuperchainConfig _superchainConfig,
+        uint256 _tryUpdateAnchorStateGas
     )
         external
         initializer
     {
-        for (uint256 i = 0; i < _startingAnchorRoots.length; i++) {
-            StartingAnchorRoot memory startingAnchorRoot = _startingAnchorRoots[i];
-            anchors[startingAnchorRoot.gameType] = startingAnchorRoot.outputRoot;
-        }
+        disputeGameFactory = _disputeGameFactory;
+        disputeGameFinalityDelaySeconds = _disputeGameFinalityDelaySeconds;
+        _anchorGame = _authorizedGame;
         superchainConfig = _superchainConfig;
+        tryUpdateAnchorStateGas = _tryUpdateAnchorStateGas;
     }
 
-    /// @notice Returns the DisputeGameFactory address.
-    /// @return DisputeGameFactory address.
-    function disputeGameFactory() external view returns (IDisputeGameFactory) {
-        return DISPUTE_GAME_FACTORY;
+    /// @notice Returns the output root of the anchor game, or an authorized anchor state if no such game exists.
+    function anchors(GameType /* unused */ ) public view returns (OutputRoot memory) {
+        if (isGameBlacklisted[_anchorGame]) revert InvalidGame();
+        return
+            OutputRoot({ l2BlockNumber: _anchorGame.l2BlockNumber(), root: Hash.wrap(_anchorGame.rootClaim().raw()) });
     }
 
-    /// @notice Callable by FaultDisputeGame contracts to update the anchor state. Pulls the anchor state directly from
-    ///         the FaultDisputeGame contract and stores it in the registry if the new anchor state is valid and the
-    ///         state is newer than the current anchor state.
-    function tryUpdateAnchorState() external {
-        // Grab the game and game data.
-        IFaultDisputeGame game = IFaultDisputeGame(msg.sender);
-        (GameType gameType, Claim rootClaim, bytes memory extraData) = game.gameData();
+    function pokeAnchorState(uint256 _candidateGameIndex) external {
```

Comment:
we didn't include this in the implementation spec yet but it was talked about in design doc, lmk what you think about it taking an `anchorGameCandidateIndex`